src/registrar/models/user.py

```diff
@@ -114,23 +145,61 @@ def needs_identity_verification(cls, email, uuid):
         except Exception as err:
             raise err
 
-        # A new incoming user who is a domain manager for one of the domains
-        # that we inputted from Verisign (that is, their email address appears
-        # in the username field of a TransitionDomain)
-        if TransitionDomain.objects.filter(username=email).exists():
-            return False
+        # We can't set the verification type here because the user may not
+        # always exist at this point. We do it down the line.
+        verification_type = cls.get_verification_type_from_email(email)
+
+        # The user needs identity verification if they don't meet
+        # any special criteria, i.e. we are validating them "regularly"
+        needs_verification = verification_type == cls.VerificationTypeChoices.REGULAR
```

Comment:
Can this be simplified to something like return verification_type == cls.VerificationTypeChoices.Regular? And we can just comment that this value indicates it needs verification as above?